src/components/Accordion/index.js

```diff
@@ -26,23 +44,75 @@ export default class Accordion extends Component {
         return null;
     }
 
-    handleToggleSection(event, name) {
+    setAsSelectAccordionSection(accordionSectionIndex) {
+        const { accordionSection } = this.state;
+        accordionSection[accordionSectionIndex].ref.focus();
+        this.setState({ currentSection: accordionSection[accordionSectionIndex].name });
+    }
+
+    handleToggleSection(event, name, currentSection) {
         const { onToggleSection } = this.props;
         if (typeof onToggleSection === 'function') {
             return onToggleSection(event, name);
         }
-        return this.setState({ activeNames: name });
+        return this.setState({ activeNames: name, currentSection });
+    }
+
+    handleKeyPressed(event) {
+        if (this.keyHandlerMap[event.keyCode]) {
+            event.preventDefault();
+            return this.keyHandlerMap[event.keyCode]();
+        }
+        return null;
+    }
+
+    selectAccordionSection(side) {
+        const { accordionSection, currentSection } = this.state;
+        const accordionSectionIndex = accordionSection.findIndex(
+            section => section.name === currentSection,
+        );
+        if (accordionSectionIndex === accordionSection.length - 1 && side === RIGHT_SIDE) {
+            this.setAsSelectAccordionSection(0);
+        } else if (accordionSectionIndex === 0 && side === LEFT_SIDE) {
+            this.setAsSelectAccordionSection(accordionSection.length - 1);
+        } else {
+            this.setAsSelectAccordionSection(accordionSectionIndex + side);
+        }
+    }
+
+    registerAccordionSection(section) {
+        const { accordionSection } = this.state;
+        const [...nodes] = getChildAccordionSectionNodes(this.containerRef.current);
+        const newChildrenRefs = insertChildOrderly(accordionSection, section, nodes);
+        this.setState({
+            accordionSection: newChildrenRefs,
+        });
+    }
+
+    unregisterAccordionSection(sectionName) {
+        const { accordionSection } = this.state;
+        const newAccordionSectionChildren = accordionSection.filter(
+            section => section.name !== sectionName,
+        );
+        this.setState({ accordionSection: newAccordionSectionChildren });
     }
 
     render() {
         const { id, children, style, className } = this.props;
         return (
             <ul
+                ref={this.containerRef}
+                onKeyDown={this.handleKeyPressed}
+                role="tablist"
                 id={id}
                 className={className}
                 style={style}>
 
-                <Provider value={this.state}>
+                <Provider value={{
+                    ...this.state,
+                    privateRegisterAccordionSection: this.registerAccordionSection,
+                    privateUnregisterAccordionSection: this.unregisterAccordionSection,
```

Comment:
again really!!!
no more js with jxs
remember separation of concerns and also you are recreating a new object in each rerender
put all this in state an then use:
```<Provider value={this.state}>```